libc/src/__support/str_to_float.h

```diff
@@ -1206,6 +1226,29 @@ LIBC_INLINE StrToNumResult<T> strtofloatingpoint(const char *__restrict src) {
   return {T(result), index, error};
 }
 
+template <class T> LIBC_INLINE StrToNumResult<T> strtonan(const char *arg) {
```

Comment:
Nit: this is internal function, expand the function name to `str_to_nan` to make it a bit easier to read.

Reply:
done